arangod/AsyncRegistryServer/Feature.cpp

```diff
@@ -56,3 +65,47 @@ auto Feature::create_metrics(arangodb::metrics::MetricsFeature& metrics_feature)
       metrics_feature.addShared(arangodb_async_threads_total{}),
       metrics_feature.addShared(arangodb_async_functions_total{}));
 }
+
+struct Feature::PromiseCleanupThread {
+  PromiseCleanupThread(size_t gc_timeout)
+      : _thread([gc_timeout, this](std::stop_token stoken) {
+          while (not stoken.stop_requested()) {
+            std::unique_lock guard(_mutex);
+            auto status = _cv.wait_for(guard, std::chrono::seconds{gc_timeout});
+            if (status == std::cv_status::timeout) {
+              async_registry::registry.run_external_cleanup();
+            }
+          }
+        }) {}
+
+  ~PromiseCleanupThread() {
+    _thread.request_stop();
+    _cv.notify_one();
+  }
+
+  std::jthread _thread;
+  std::mutex _mutex;
+  std::condition_variable _cv;
```

Comment:
The order of members here is wrong. `std::jthread` has to be last, so that it is destroyed first. `std::jthread` waits for the thread to join on destruction. Otherwise `_mutex` might already be destroyed but then accessed within `_thread`. Sorry, my bad. I wrote that code. 😞